File: num10.py
prime_number = list(range(0,2000001))
sum = 0

for i in range(2000001):
    prime_number[i] = i

for i in range(2, 2000001):
    if prime_number[i] == 0 :
        continue
    for j in range(i+i,2000001,i):
        prime_number[j] = 0

for i in range(2,2000001):
    if prime_number[i] != 0 :
        sum += prime_number[i]

print(sum)

File: num10.c
//The sum of the primes below 10 is 2 + 3 + 5 + 7 = 17.
//Find the sum of all the primes below two million.

#include <stdio.h>

int main()                            // Sieve of Eratosthenes
{
    long long sum=0;
    int prime_number[2000001];
    
    for(int i=0; i<=2000000; i++)
    {
        prime_number[i]=i;        
    }
    
    for(int i=2; i <=2000000; i++)
    {
        if(prime_number[i] == 0)
        {
            continue;
        }
        
        for(int j=i+i; j <= 2000000; j += i)
        {
            prime_number[j] = 0;
        }
     }
     
     for(int i=2; i <= 2000000; i++)
     {
         if(prime_number[i] != 0)
         {
             sum += prime_number[i];
         }
     }
    
    printf("%llu\n", sum);
    return 0;
}

// answer : 142913828922

